app/models/distributed_valid_products.rb

```diff
@@ -0,0 +1,41 @@
+# Finds valid products distributed by a particular distributor in an order cycle
+#
+# If a product without variants is added to an order cycle, and then some
+# variants are added to that product, but not the order cycle, then the master
+# variant should not available for customers to purchase. This class filters
+# out such products so that the customer cannot purchase them.
+class DistributedValidProducts
+  def initialize(order_cycle, distributor)
```

Comment:
IMO, this should be in app/services, it's a service and it's doing one specific thing really well, finding the order cycle distributed products. I think it deserves Order Cycle in the name.   Order_Cycle_Distributed_Products?

Reply:
OrderCycleProductsFinder.call
FindOrderCycleProducts.call
please cast you vote here ❤️  https://community.openfoodnetwork.org/t/how-to-name-a-new-service-in-ofn/1530/5

Reply:
lol. I first started adding it into models and didn't put any thought on it afterward. Do you think is worth keeping the notion of validation in its name? It felt important to me to know that we perform a sort of validation in here but what do you guys think?

Reply:
ehehe, I'd not include Valid because the validation isn't valid 🤣 
Because we have the standard variant, master variants should never be added to OCs or exchanges, so this validation should go... doesnt need to be now, but it should go at some point.

Reply:
done